tests/unit/config/test_configtypes.py

```diff
@@ -374,19 +387,60 @@ def test_validate_invalid(self, klass, val):
 
     def test_invalid_empty_value_none_ok(self, klass):
         with pytest.raises(configexc.ValidationError):
-            klass(none_ok=True).validate('foo,,bar')
+            klass(none_ok_outer=True).validate('foo,,bar')
+            klass(none_ok_inner=True).validate('')
 
     @pytest.mark.parametrize('val, expected', [
         ('foo', ['foo']),
         ('foo,bar,baz', ['foo', 'bar', 'baz']),
         ('', None),
-        # Not implemented yet
-        pytest.mark.xfail(('foo, bar', ['foo', 'bar'])),
+        ('foo, bar', ['foo', 'bar'])
     ])
     def test_transform(self, klass, val, expected):
         assert klass().transform(val) == expected
 
 
+class LengthListSubclass(configtypes.LengthList):
+
+    """A subclass of LengthList which we use in tests."""
+
+    def __init__(self, length=None, none_ok_inner=False, none_ok_outer=False):
+        super().__init__(configtypes.Int(none_ok=none_ok_inner),
+                         length=length, none_ok=none_ok_outer)
+
+
+class TestLengthList:
+
+    """Test List with length."""
+
+    @pytest.fixture
+    def klass(self):
+        return LengthListSubclass
+
+    @pytest.mark.parametrize('val', ['', '0,1,2', '-5,4,2'])
+    def test_validate_valid(self, klass, val):
+        klass(none_ok_outer=True, length=3).validate(val)
+
+    @pytest.mark.parametrize('val', ['', '1,,4'])
+    def test_validate_invalid(self, klass, val):
+        with pytest.raises(configexc.ValidationError):
+            klass().validate(val)
+
+    def test_invalid_empty_value_none_ok(self, klass):
+        with pytest.raises(configexc.ValidationError):
+            klass(none_ok_outer=True).validate('1,,4')
```

Comment:
Oh, right, I missed that. Then this should be two `pytest.raises` blocks though - otherwise only the first line is run. The second will never run, as the first one raises an exception.